### PR TITLE
feat(components/pages): add `sky-link-list` component

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#3033904](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3033904) [AB#3033919](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3033919)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2784/storybook/)

Component Storybooks:

- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2784/storybooks/indicators-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2784/storybooks/pages-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2784/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2784/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2784/playground/)

File: apps/e2e/pages-storybook/src/app/shared/links/links.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { SkyLinkListModule } from '@skyux/pages';
+
+@Component({
+  selector: 'app-links',
+  standalone: true,
+  imports: [SkyLinkListModule],
+  templateUrl: './links.component.html',
+  styles: `
+    :host {
+      display: contents;
```
Comment (Blackbaud-SteveBrush) [inline apps/e2e/pages-storybook/src/app/shared/links/links.component.ts]: Would we expect our consumers to use `display: contents` if they wish to share links between multiple pages?

File: libs/components/pages/src/lib/modules/link-list/link-list-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Component } from '@angular/core';
+
+/**
+ * A wrapper for each link in a link list.
+ * @internal
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/link-list/link-list-item.component.ts]: The story mentions adding the link components to the "page" docs. Are you keeping this internal until everything's ready to go?

File: libs/components/pages/src/lib/modules/link-list/link-list.module.ts [hunk 3]
```diff
@@ -1,23 +1,10 @@
...
-  imports: [
-    CommonModule,
-    SkyAppLinkModule,
-    SkyHrefModule,
-    SkyThemeModule,
-    SkyWaitModule,
-    LinkAsModule,
-  ],
+  exports: [SkyLinkListComponent, SkyLinkListItemComponent],
+  imports: [SkyLinkListComponent, SkyLinkListItemComponent],
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/link-list/link-list.module.ts:8]: The story mentions exporting the `SkyLinkListModule` from the `SkyPageModule`. Is that feasible?

File: libs/components/pages/src/lib/modules/link-list/link-list.component.ts [hunk 4]
```diff
@@ -1,28 +1,52 @@
...
 import { SkyPageLink } from '../action-hub/types/page-link';
 import { SkyPageLinksInput } from '../action-hub/types/page-links-input';
+import { LinkAsModule } from '../link-as/link-as.module';
 
+import { SkyLinkListItemComponent } from './link-list-item.component';
+
+/**
+ * A component that displays a list of links on the right side of the page, or
+ * below the page content on mobile devices.
+ * @internal
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/link-list/link-list.component.ts]: Same here, `@internal`.

File: apps/e2e/pages-storybook/src/app/shared/links/links.component.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    <a href="#">Link 1</a>
+  </sky-link-list-item>
+  <sky-link-list-item>
+    <a href="#">Link 2</a>
+  </sky-link-list-item>
+  <sky-link-list-item>
+    <a href="#">Link 3</a>
+  </sky-link-list-item>
+  <sky-link-list-item>
+    <a href="#">Link 4</a>
```
Comment (Blackbaud-SteveBrush) [inline apps/e2e/pages-storybook/src/app/shared/links/links.component.html]: Buttons should also be supported with this work. Can we throw a `button` in one of these to confirm the styles are appropriate in the screenshot?

File: libs/components/pages/testing/src/link-list/link-list-harness.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+import { SkyWaitHarness } from '@skyux/indicators/testing';
+
+import { SkyLinkListHarnessFilters } from './link-list-harness-filters';
+
+/**
+ * Harness for interacting with a link list component in tests.
+ * @internal
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/testing/src/link-list/link-list-harness.ts]: Same here, `@internal`?

File: apps/e2e/pages-storybook/src/app/shared/links/links.component.ts (hunk 1 above)
Comment (johnhwhite) [reply apps/e2e/pages-storybook/src/app/shared/links/links.component.ts]: It wasn't needed, I think I was checking something else. I've removed the extra styles.

File: libs/components/pages/src/lib/modules/link-list/link-list-item.component.ts (hunk 2 above)
Comment (johnhwhite) [reply libs/components/pages/src/lib/modules/link-list/link-list-item.component.ts]: The `@internal` isn't really needed. Removed.

File: libs/components/pages/src/lib/modules/link-list/link-list.component.ts (hunk 4 above)
Comment (johnhwhite) [reply libs/components/pages/src/lib/modules/link-list/link-list.component.ts]: The `@internal` isn't really needed. Removed.

File: apps/e2e/pages-storybook/src/app/shared/links/links.component.html (hunk 5 above)
Comment (johnhwhite) [reply apps/e2e/pages-storybook/src/app/shared/links/links.component.html]: Added a button example.

File: libs/components/pages/testing/src/link-list/link-list-harness.ts (hunk 6 above)
Comment (johnhwhite) [reply libs/components/pages/testing/src/link-list/link-list-harness.ts]: The `@internal` isn't really needed. Removed.

File: libs/components/pages/src/lib/modules/link-list/link-list.module.ts (hunk 3 above)
Comment (johnhwhite) [reply libs/components/pages/src/lib/modules/link-list/link-list.module.ts:8]: Good catch. I've added the new components to `SkyPageModule`'s exports.

File: libs/components/pages/src/lib/modules/link-list/link-list.component.ts [hunk 7]
```diff
@@ -1,28 +1,51 @@
...
-  public set links(value: SkyPageLinksInput | undefined) {
-    this.#_links = value;
-    this.linksArray = Array.isArray(value) ? value : [];
-  }
-
-  public get links(): SkyPageLinksInput | undefined {
-    return this.#_links;
-  }
+  public readonly headingText = input<string>();
+  public readonly links = input<SkyPageLinksInput | undefined>();
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/link-list/link-list.component.ts:38]: These two inputs should have JSDocs.

Comment (johnhwhite) [reply libs/components/pages/src/lib/modules/link-list/link-list.component.ts:38]: Good call. Added.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**